### PR TITLE
Only run test_issue_223 when deflate filter is avail

### PR DESCRIPTION
**Pull request by mulimoen**
A meta-PR targeting the branch of #246

### CODE REVIEW

File: hdf5-sys/build.rs [hunk 1]
```diff
@@ -204,6 +205,7 @@ impl Header {
                 hdr.have_threadsafe = value > 0;
             } else if name == "H5_HAVE_FILTER_DEFLATE" {
                 hdr.have_zlib = value > 0;
+                hdr.have_filter_deflate = value > 0;
```
Comment (aldanor) [inline hdf5-sys/build.rs:208]: Ok, I'm somewhat confused, in HDF5 terminology isn't "deflate" = "zlib"?

Comment (mulimoen) [reply hdf5-sys/build.rs:208]: They are the mostly the same and used interchangeably. The compile definition is `HDF5_ENABLE_Z_LIB_SUPPORT` but the flag in `H5pubconf.h` is `H5_HAVE_FILTER_DEFLATE`.

Comment (mulimoen) [reply hdf5-sys/build.rs:208]: I don't mind having just `have_zlib` or `have_filter_deflate`

Comment (aldanor) [reply hdf5-sys/build.rs:208]: Hm, just a single `have_filter_deflate` is probably more correct since that's the flag we're checking in `H5pubconf`?

**Comment by aldanor** [comment]
👍 I'll merge it into the f16 branch and check the builds there

**Pull request merged**